### PR TITLE
chore: upgrade istio from 1.8.1 to 1.8.2

### PR DESCRIPTION
**Pull request by GoelDeepak**
**What type of PR is this?**
Chore 

**What this PR does/ why we need it**:
Bump istio from 1.8.1 to 1.8.2

**Which issue(s) this PR fixes**:
https://github.com/istio/istio/issues/29566

**Special notes for your reviewer**:
This PR brings a fix for the issue that blocks Kaptain from running on istio-1.8.1

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by akirillov — APPROVED** [review]
Thanks, @GoelDeepak. I can confirm that the upgrade to Istio 1.8.2 fixes the issue from the ticket. LGTM

**Review by gracedo — APPROVED** [review]

File: staging/istio/templates/deployment.yaml [hunk 1]
```diff
@@ -33,12 +33,7 @@ spec:
             runAsNonRoot: true
           imagePullPolicy: IfNotPresent
           resources:
-            limits:
```
Comment (vespian) [inline staging/istio/templates/deployment.yaml:36]: Why?

Comment (hectorj2f) [reply staging/istio/templates/deployment.yaml:36]: Yeah, removing the requests is not a good practice, and certain k8s benchmarks could report it as a WARNING or ERROR. Please, justify why you removed the limits, and revert the requests.

Comment (hectorj2f) [reply staging/istio/templates/deployment.yaml:36]: oh :D, I just saw that their in the values.yaml 🙄 , all good :)

**Review by hectorj2f — APPROVED** [review]

**Review by vespian — APPROVED** [review]

**Pull request merged**